circuits/circuits/register/register_aadhaar.circom

```diff
@@ -0,0 +1,92 @@
+pragma circom 2.1.9;
+
+include "circomlib/circuits/bitify.circom";
+include "circomlib/circuits/poseidon.circom";
+include "../utils/aadhaar/extractQrData.circom";
+include "../utils/aadhaar/signature.circom";
+include "../utils/passport/customHashers.circom";
+include "@openpassport/zk-email-circuits/utils/array.circom";
+
+/// @title: AadhaarRegister
+/// @notice Main circuit — verifies the integrity of the aadhaar data, the signature, and generates commitment and nullifier
+/// @param n RSA pubic key size per chunk
+/// @param k Number of chunks the RSA public key is split into
+/// @param maxDataLength Maximum length of the data
+/// @input qrDataPadded QR data without the signature; assumes elements to be bytes; remaining space is padded with 0
+/// @input qrDataPaddedLength Length of padded QR data
+/// @input delimiterIndices Indices of delimiters (255) in the QR text data. 18 delimiters including photo
+/// @input signature RSA signature split into k chunks of n bits each
+/// @input pubKey RSA public key(of the government) split into k chunks of n bits each
+/// @input secret Secret for commitment generation. Saved by the user to access their commitment
+/// @input attestation_id Attestation ID of the credential used to generate the commitment
+/// @output nullifier Generated nullifier - deterministic on the Aadhaar data
+/// @output commitment Commitment that will be added to the onchain registration tree
+template REGISTER_AADHAAR(n, k, maxDataLength){
+
+    signal input qrDataPadded[maxDataLength];
+    signal input qrDataPaddedLength;
+    signal input delimiterIndices[18];
+    signal input pubKey[k];
+    signal input signature[k];
+
+    signal input secret;
+    // Aadhaar = 6
+    signal input attestation_id;
+
+
+    // Assert `qrDataPaddedLength` fits in `ceil(log2(maxDataLength))`
+    component n2bHeaderLength = Num2Bits(log2Ceil(maxDataLength));
+    n2bHeaderLength.in <== qrDataPaddedLength;
+
+    // Verify the RSA signature
+    component signatureVerifier = SignatureVerifier(n, k, maxDataLength);
```

Comment:
https://github.com/selfxyz/self/blob/main/circuits/circuits/utils/passport/signatureVerifier.circom

can you use this?